e2e_context_test.go

```diff
@@ -393,3 +393,45 @@ func TestContextPerRequestWithTimeoutCancelledByContext(t *testing.T) {
 	// expected error should occur
 	assert.True(t, suppressor.expErrorOccurred)
 }
+
+func TestContextPerRequestRetryCancelledByContext(t *testing.T) {
+	var callCount int
+	var isConfigCtxCancelled bool
+	delayDuration := TimeOutDuration / 2
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		callCount++
```

Comment:
Since handler is invoked from another goroutine, we should use atomic for call count.